schema/datatypes/hardware_info.fbs

```diff
@@ -24,6 +24,23 @@ enum ImuType: uint16 {
     ICM20948,
 }
 
+enum BoardType: uint16 {
```

Comment:
We should think carefully about what we define a board here to mean. Originally we represented it as a string because it was supposed to be diagnostic information self reported by firmware. Are we using the enum somewhere to do different things based on the board?

I think that it may be a fools errand to try to have a fixed set of boards, since there are an infinite number of DIY boards with different pins, different microcontroller footprints, etc. Is it possible to just keep this as a string?

Reply:
With the comming firmware tool i need a way to know the board specified in the firmware as i use it to prevent mistakes when using the tool. 

IF you try to update a tracker using ota, only the devices that match the the built firmware will be allowed to be flashed.

Reply:
i agree with your concern but i also do not see any better path forward at the moment, one thing to consider is that the firmware udp protocol is also using the same enum inside the server to generate the string i just deprecated. 
The string was just a weak representation of the actual enum sent by the firmware protocol and is curently the only way to know what board the firmware is set to

Reply:
perhaps we could have a fixed set of "official" variants which are in this enum, as well as an "other" variant which would then be described as a string?

Reply:
i am fine with the secondary string for unknown variants. but i would like to keep the official enum the same as the firmware for easier mapping and consistency. lets not change everything please